WcaOnRails/app/controllers/users_controller.rb

```diff
@@ -128,13 +110,20 @@ def update
   end
 
   private def user_params
-    user_params = params.require(:user).permit(*current_user.editable_fields_of_user(user_to_edit))
+    permitted_params = current_user.editable_fields_of_user(user_to_edit).to_a
+    permitted_params.delete :preferred_event_ids
```

Comment:
I've added the `preferred_event_ids` in `user#editable_fields_for_user`. It allow us to use `disabled: !editable_fields.include?(:preferred_event_ids)` as for other fields. But then we want to ensure that only valid events are passed via params so we replace it with this `preferred_event_ids: Event.all_official.map(&:id)` here. Do you see any cleaner way?